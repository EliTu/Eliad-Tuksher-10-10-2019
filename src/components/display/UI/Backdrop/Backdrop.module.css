.BackdropStyles {
    width: 100%;
    height: 100%;
    background-color: rgba(15, 15, 15, 0.541);
}