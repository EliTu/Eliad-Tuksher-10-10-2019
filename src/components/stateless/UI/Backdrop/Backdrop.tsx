import React from 'react';
import styles from './Backdrop.module.css';

interface IProps {
	isDisplayed: boolean;
}

const Backdrop: React.FC<IProps> = ({ isDisplayed }) => {
	const { BackdropStyles } = styles;

	return <>{isDisplayed && <div className={BackdropStyles}></div>}</>;
};

export default Backdrop;
