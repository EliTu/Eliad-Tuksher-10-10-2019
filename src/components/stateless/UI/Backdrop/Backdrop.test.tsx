import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Backdrop from './Backdrop';

describe('Backdrop component', () => {
	let component: ShallowWrapper;

	beforeEach(() => {
		component = shallow(<Backdrop isDisplayed />);
	});

	it('should render without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('if isDisplayed props is true, should have a div with the className of BackdropStyles', () => {
		const div = component.find('div');

		expect(div.length).toBe(1);
		expect(div).toHaveClassName('BackdropStyles');
	});

	it('if isDisplayed prop is false, should not have a div with the BackdropStyles', () => {
		component.setProps({ isDisplayed: false });
		const div = component.find('div');

		expect(div.length).toBe(0);
		expect(div.length).not.toBe(1);
	});
});
