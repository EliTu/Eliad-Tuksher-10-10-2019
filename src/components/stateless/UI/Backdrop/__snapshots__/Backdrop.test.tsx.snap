// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop component should render without errors 1`] = `
<Fragment>
  <div
    className="BackdropStyles"
  />
</Fragment>
`;
